webapp/src/components/systemConsole/expiry/expiry.tsx

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import type {ChangeEvent} from 'react';
+import React, {useCallback, useEffect, useState} from 'react';
+
+import type {CustomComponentProps, ExpiryConfig} from 'types/mattermost-webapp';
+
+import './style.scss';
+
+const defaultExpiry = '30';
+
+const Expiry = ({id, setSaveNeeded, onChange, config}: CustomComponentProps) => {
+    const [expiryDays, setExpiryDays] = useState<string>(defaultExpiry);
+    const [error, setError] = useState<string>('');
+
+    useEffect(() => {
+        // Set initial value from saved config
+
+        const expiryConfig = config.PluginSettings?.Plugins?.['com.mattermost.user-survey']?.surveyexpiry;
```

Comment:
`com.mattermost.user-survey` do you want to extract this as well as a constant

Reply:
Maybe we can use `manifest.id` for this one?

Reply:
The key is part of type definition in `webapp/src/types/mattermost-webapp/index.d.ts` so can't be fudged without changing the type declaration.